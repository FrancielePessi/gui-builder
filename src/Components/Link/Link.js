import React from 'react';
import PropTypes from 'prop-types';
import Link from '@material-ui/core/Link';

const LinkButton = ({ active = false, children, onClick }) => {
    if (active) {
        return (
        <span
            style={{
                marginLeft: '6px',
                marginRight: '6px',
            }}
        >
            {children}
        </span>);
    }

    return (
        <Link
            component="button"
            onClick={onClick}
            disabled={active}
            style={{
                marginLeft: '6px',
                marginRight: '6px',
                textDecoration: 'underline',
            }}
        >
            {children}
        </Link>
    );
};

Link.propTypes = {
    active: PropTypes.bool.isRequired,
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func.isRequired,
}

export default LinkButton;
