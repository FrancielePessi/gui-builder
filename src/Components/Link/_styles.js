const styles = (theme) => ({
    activeItem: {
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        textDecoration: 'underline',
    },
});

export default styles;
