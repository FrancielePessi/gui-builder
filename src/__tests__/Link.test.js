import React from 'react';
import renderer from 'react-test-renderer';

import Link from '../Components/Link/Link';

it('Link with href attribute should render correctly', () => {
    const tree = renderer
        .create(<Link href="whatever" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
