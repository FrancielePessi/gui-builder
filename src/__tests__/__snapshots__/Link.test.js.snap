// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-activeItem-1 MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;

exports[`Link with href attribute should render correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-activeItem-22 MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
